### ISSUE TITLE
GenerateManifest in workbox-webpack-plugin

### ISSUE THREAD

**Issue by ujjwalmishra09** [GoogleChrome/workbox#1485]
**Library Affected**:
*workbox-webpack-plugin*

**Issue or Feature Request Description**:
Like `workbox-webpack-plugin~InjectManifest`, can we have  `workbox-webpack-plugin~GenerateManifest`? 

It should have one callback function with following arguments.
1. webpack plugin callbacks arguments (`compiler`, `compilation`)
2. `generated-precahce-manifest-filename`

Now developer can inject precahce-manifest-filename in compiled service worker file according to his choice.

**Comment by jeffposnick** [comment]
Thanks for the request!

Can you share some more background on what benefits that would offer vs. using `InjectManifest`, which does allow you to add the manifest information to what could potentially be a compiled service worker file? Perhaps there are some improvements to the `InjectManifest` use case that would make it easier to accomplish what you're looking to do, without having to add support for a brand-new mode.

**Comment by ujjwalmishra09** [comment]
```
var config = {
     entry: {
         'service-worker': './src/js/service-worker.js'
     },
     output: {
         path: path.resolve(__dirname, buildOutputPath),
         publicPath: publicHost + '/static/',
         filename: './js/[name].[chunkhash].js',
         chunkFilename: './js/chunk.[chunkhash].js'
     },
     plugins: [
        new WorkboxPlugin.InjectManifest({
            swSrc: './src/js/service-worker.js'
        })
     ]
 }
```

In above config, I am getting two service worker file.
1. Generated by webpack with hash key (minified service worker)
2. Generated by webpack-workbox-plugin
i don't want to use 2nd one as service worker but want to inject `precahce-manifest` file name in first one.

**Comment by Jamiewarb** [comment]
Try specifying a `swDest` as well as an `swSrc`. Then make sure webpack is only minifying the destination file, and not the source file. The source will have the manifest injected into it then stored in the destination file, which can then be minified or whatever with webpack.

E.g. my config is:

```
new WorkboxPlugin.InjectManifest({
    swSrc: `service-worker-template.js`,
    swDest: `sw.js`,
    importsDirectory: path.join(__dirname, 'public/precache'),
    include: [/\.css$/, /\.js$/],
})
```

This grabs a template SW file, where I've written all of my service worker logic, called service-worker-template.js. It then imports the workbox library and my manifest file, and outputs both this and the precache manifest to my `/public` directory. Workbox can then do what it wants with it there

**Comment by jeffposnick** [comment]
In Workbox webpack plugin v5, the `swSrc` you provide to `InjectManifest` will be compiled by webpack. I think that takes care of your use case.

**Comment by jeffposnick** [comment]
This should be addressed by the current Workbox v5.0.0 alpha.
